lib/exception_handling.rb

```diff
@@ -177,26 +177,15 @@ def log_error(exception_or_string, exception_context = '', controller = nil, tre
         exception_info = ExceptionInfo.new(ex, exception_context, timestamp, controller || current_controller, data_callback)
 
         if stub_handler
-          return stub_handler.handle_stub_log_error(exception_info.data)
-        end
-
-        write_exception_to_log(ex, exception_context, timestamp)
-        execute_custom_log_error_callback(exception_info.enhanced_data, exception_info.exception, treat_like_warning)
-
-        if treat_like_warning  #Bail out before sending to honeybadger or email
-          return
-        end
-
-        if !ex.is_a?(Warning)
-          if honeybadger?
-            send_exception_to_honeybadger(exception_info)
-          end
-
-          if should_send_email?
-            log_error_email(exception_info)
-          end
+          stub_handler.handle_stub_log_error(exception_info.data)
+        else
+          write_exception_to_log(ex, exception_context, timestamp)
+          external_notification_results = unless treat_like_warning || ex.is_a?(Warning)
+                                            send_external_notifications(exception_info)
+                                          end || {}
+          execute_custom_log_error_callback(exception_info.enhanced_data, exception_info.exception, treat_like_warning, external_notification_results)
```

Comment:
Are we changing the callbacks for `web` and `PNAPI` as part of this branch? And are those the only two repos that use the callback? Need to be careful to communicate this change out if not to get ahead of errors if people update their exception_handling version.

Reply:
Correct, I'm changing the callbacks for `web` and `pnapi`. I'm not sure what other repos may use the callback. Perhaps I should make it backwards compatible by making the new arg optional. Would that be sufficient?

Reply:
Is there a way to pass in a fourth argument but make it optional if the method only takes 3? I'm not sure the backwards compatibility can be built in on this side...

Reply:
I can check the `arity` of the callback method and only pass through the matching number of arguments.

Reply:
I'm pretty sure only `web` and `PNAPI` use it, and other repos/gems would need to update the `exception_handling` gem SHA to run into it, so I don't think it's worth it...

Doing a search for `post_log_error_hook` did lead me to the README though, which should be updated with the latest params:  https://github.com/Invoca/exception_handling/blob/master/README.md#post_log_error_hook

Reply:
Something like this?
```
if ExceptionHandling.post_log_error_hook
  honeybadger_status = external_notification_results[:honeybadger_status] || :skipped
  params = [exception_data, exception, treat_like_warning, honeybadger_status].first(ExceptionHandling.post_log_error_hook.arity.abs)
  ExceptionHandling.post_log_error_hook.call(*params)
end
```